frontend/src/components/summary/PieChart.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useRef } from 'react';
+import * as d3 from 'd3';
+import tippy from 'tippy.js';
+import piechartStyle from './piechart.module.scss';
+
+interface JobTimes {
+  [key: string]: number;
+}
```

Comment:
Can be generic (as with PieChartData). I would have 
```
export interface Dictionary<T> {
  [key: string]: T;
}
```
in somewhere like `model/common/dictionary.ts` (And have index.ts files throughout model folder).

Reply:
This looks like such a common thing to use that I'm surprised that typescript doesn't provide this out of the box!